File: app/src/types/request.ts
export interface WebhookFile {
  name: string;
  contentType: string;
  size: number;
  data: string; // base64 encoded
}

export interface WebhookRequest {
  id: string;
  timestamp: string;
  method: string;
  path: string;
  headers: Record<string, string[]>;
  body: string;
  queryParams: Record<string, string[]>;
  files?: WebhookFile[];
  formFields?: Record<string, string>;
}

File: api/models/request.go
package models

import "time"

type File struct {
	Name        string `json:"name"`
	ContentType string `json:"contentType"`
	Size        int64  `json:"size"`
	Data        string `json:"data"` // base64 encoded
}

type Request struct {
	ID          string              `json:"id"`
	Timestamp   time.Time           `json:"timestamp"`
	Method      string              `json:"method"`
	Path        string              `json:"path"`
	Headers     map[string][]string `json:"headers"`
	Body        string              `json:"body"`
	QueryParams map[string][]string `json:"queryParams"`
	Files       []File              `json:"files,omitempty"`
	FormFields  map[string]string   `json:"formFields,omitempty"`
}
